app/src/main/java/de/westnordost/streetcomplete/ktx/Element.kt

```diff
@@ -1,36 +1,35 @@
 package de.westnordost.streetcomplete.ktx
 
-import de.westnordost.osmapi.map.data.*
-import de.westnordost.osmapi.map.data.Element.Type.*
 import de.westnordost.osmfeatures.GeometryType
 import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.mapdata.*
 import java.util.*
-import kotlin.collections.HashMap
 
-fun Element.copy(newId: Long = id, newVersion: Int = version, newDateEdited: Date? = dateEdited): OsmElement {
-    val tags = tags?.let { HashMap(it) }
+fun Element.copy(
+    newId: Long = id,
+    newVersion: Int = version,
+    newTimestampEdited: Long = timestampEdited,
+    newTags: Map<String, String> = tags,
+): Element {
     return when (this) {
-        is Node -> OsmNode(newId, newVersion, position, tags, null, newDateEdited)
-        is Way -> OsmWay(newId, newVersion, ArrayList(nodeIds), tags, null, newDateEdited)
-        is Relation -> OsmRelation(newId, newVersion, ArrayList(members), tags, null, newDateEdited)
-        else -> throw RuntimeException()
+        is Node -> Node(newId, position, newTags, newVersion, newTimestampEdited)
+        is Way -> Way(newId, ArrayList(nodeIds), newTags, newVersion, newTimestampEdited)
+        is Relation -> Relation(newId, ArrayList(members), newTags, newVersion, newTimestampEdited)
     }
 }
```

Comment:
All data classes in Kotlin already have a `copy` method, so this is superfluous now.

Reply:
But `Element` is not a data class. At least in one place, `copy` is called on an `Element`, so we'd then need to differentiate between `Node`/`Way`/`Relation` there.

Reply:
Ah, ok